cli/src/program.rs

```diff
@@ -2494,6 +2538,11 @@ fn do_process_program_upgrade(
             )
         };
 
+        // Add extend program instruction if ProgramData account needs to be extended
+        if let Some(add_program_extend_ix) = add_program_extend_ix {
+            initial_instructions.push(add_program_extend_ix);
+        }
+
```

Comment:
Any particular reason why you don't just slide that whole block down here, and only add the instruction if it resolves to what you currently have as this?

```rust
Some(bpf_loader_upgradeable::extend_program(
    program_id,
    Some(&fee_payer_signer.pubkey()),
    additional_bytes as u32,
))
```

Reply:
thought we could save some function calls if i have checks for program data at the top